### PR TITLE
autotools v2 plugin: adding cross compilation support

### PR DESCRIPTION
**Pull request by kubiko**
Adding minimal cross compilation support for autotools v2 plugin

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

**Review by cjp256 — COMMENTED** [review]
To best support this, we need to add a cross compilation variant of the hello-world test. (tests/spread/plugins/v2/)

**Comment by kubiko** [comment]
@cjp256 I changed the name of the keyword.
What's the location of hello-world snap? one was core18 works with cross compilation, so if only base was updated for core20, it should still work.

File: tests/unit/plugins/v2/test_autotools.py [hunk 1]
```diff
@@ -53,14 +59,15 @@ def test_get_build_environment():
...
+        autotools_configure_environment = list()
 
     plugin = AutotoolsPlugin(part_name="my-part", options=Options())
 
     assert plugin.get_build_commands() == [
         "[ ! -f ./configure ] && [ -f ./autogen.sh ] && env NOCONFIGURE=1 ./autogen.sh",
         "[ ! -f ./configure ] && [ -f ./bootstrap ] && env NOCONFIGURE=1 ./bootstrap",
         "[ ! -f ./configure ] && autoreconf --install",
-        "./configure",
+        "CC=${SNAPCRAFT_ARCH_TRIPLET}-gcc ./configure --host=${SNAPCRAFT_ARCH_TRIPLET}",
```
Comment (cjp256) [inline tests/unit/plugins/v2/test_autotools.py:70]: More generically, this should perhaps be:

```
./configure --build "$(/usr/share/libtool/build-aux/config.guess)" --host "$SNAPCRAFT_ARCH_TRIPLET"
```

With that autotools should find the correct compiler, linker, etc.

Comment (cjp256) [reply tests/unit/plugins/v2/test_autotools.py:70]: And snapcraft should export the build-machines arch triplet for convenience...  perhaps SNAPCRAFT_BUILD_ARCH_TRIPLET?

Comment (cjp256) [reply tests/unit/plugins/v2/test_autotools.py:70]: https://www.gnu.org/software/automake/manual/html_node/Cross_002dCompilation.html

**Comment by sergiusens** [comment]
Does this not move away from our prescribed build environment work? What happens if CC is not gcc?

**Comment by sergiusens** [comment]
I am going to close this, we should perhaps discuss further on the forum

**Pull request closed without merging**